diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/KdocFormatting.kt

```diff
@@ -321,4 +349,43 @@ class KdocFormatting(private val configRules: List<RulesConfig>) : Rule("kdoc-fo
     private fun ASTNode.applyToPrevSibling(elementType: IElementType, consumer: ASTNode.() -> Unit) {
         prevSibling { it.elementType == elementType }?.apply(consumer)
     }
+
+    /**
+     * Checks whether this tag's content represents an invalid version
+     */
+    private fun KDocTag.hasInvalidVersion(): Boolean {
+        val content = getContent().trim()
+        if (' ' in content || '/' in content) {
+            // Filter based on symbols that are not allowed in versions. Assuming that people put either version or date in `@since` tag.
+            return true
+        }
+        return versionRegex?.matches(content)?.not()
+            ?: dateFormats.any {
+                // try to parse, get year and check it's value sanity
+                // otherwise it might be a tricky version format
+                runCatching {
+                    it.parse(content).get(ChronoField.YEAR) > MINIMUM_SANE_YEAR
+                }
+                    .getOrNull() == true
+            }
+    }
+
+    /**
+     * A [RuleConfiguration] for KDoc formatting
+     */
+    class KdocFormatConfiguration(config: Map<String, String>) : RuleConfiguration(config) {
+        /**
+         * Regular expression, if present, against which a version should be matched in `@since` tag.
+         */
+        val versionRegex: Regex? by lazy {
+            config["versionRegex"]?.let { Regex(it) }
+        }
+    }
+
+    companion object {
+        private const val MINIMUM_SANE_YEAR = 1900
+        val dateFormats: List<DateTimeFormatter> = listOf("yyyy-dd-mm", "yyyy-mm-dd", "yyyy.mm.dd", "yyyy.dd.mm").map {
```

Comment:
Can there be format where year present by two numbers?(yy-mm-dd)

Reply:
Yes, I'll add, with `-` it should not interfere with common version patterns. Added.